app/components/visibility_badge_component.rb

```diff
@@ -2,36 +2,57 @@
 
 class VisibilityBadgeComponent < ApplicationComponent
   # @param [Work, SolrDocument]
-  def initialize(work:)
+  def initialize(work:, mini: false)
     @work = work
+    @mini = mini
   end
 
   def render
-    work.respond_to?(:visibility)
+    work.respond_to?(:visibility) && work.respond_to?(:embargoed?)
   end
 
   private
 
     attr_reader :work
 
+    def color
+      if show_embargoed?
+        'red'
+      else
+        details[visibility][:color]
+      end
+    end
+
     def html_class
       [
         'badge',
         'badge--icon',
-        "badge--icon-#{details[visibility][:color]}"
+        "badge--icon-#{color}"
       ].join(' ')
     end
 
+    def show_embargoed?
+      @mini && work.embargoed?
+    end
+
     def visibility
       work.visibility || Permissions::Visibility::PRIVATE
     end
 
     def label
-      details[visibility][:label]
+      if show_embargoed?
+        i18n_label('embargoed')
+      else
+        details[visibility][:label]
+      end
     end
 
     def icon
-      details[visibility][:icon]
+      if show_embargoed?
+        'lock_clock'
+      else
+        details[visibility][:icon]
+      end
     end
```

Comment:
I think you should be able to keep icon, label, and html_class methods the same by using the visibility and details methods:

``` ruby
def visibility
  'embargo' if work.embargoed?

  work.visibility || Permissions::Visibility::PRIVATE 
end

def details
  {
    'embargo' => {
      label: i18n_label('embargoed'),
      color: 'red',
      icon: 'lock_clock'
...
```

We are conflating visibility and embargo here, which is probably ok within the context of just this one component. However, I could also renaming the visibility method to `details_key` or just `key` for clarity.